addons/mail/static/src/models/attachment_image.js

```diff
@@ -8,6 +8,18 @@ registerModel({
     name: 'AttachmentImage',
     identifyingFields: ['attachmentList', 'attachment'],
     recordMethods: {
+        /**
+         * Called when clicking on download icon.
+         *
+         * @param {MouseEvent} ev
+         */
+        onClickDownload(ev) {
+            ev.stopPropagation(); // prevents from opening viewer
+            if (!this.attachment) {
```

Comment:
`attachment` is a mandatory field, because this is a required identifying field.
What you need is `!this.exists(0` instead.